catalog/validations/signup-validation.js

```diff
@@ -0,0 +1,45 @@
+const { body } = require('express-validator');
+
+const User = require('../models/user');
+
+const signupValidationRules = () => {
+  return [
+    body('email')
+      .trim()
+      .not()
+      .isEmpty()
+      .withMessage('Email cannot be empty.')
+      .isEmail()
+      .withMessage('Please enter a valid email.')
+      .custom((value, { req }) => {
+        return User.findOne({ email: value })
+          .then(userDoc => {
+            if (userDoc) {
+              return Promise.reject('This email is already taken.');
+            }
+          });
+      }),
+    body('username')
+      .trim()
+      .not()
+      .isEmpty()
+      .withMessage('Username cannot be empty.')
+      .custom((value, { req }) => {
+        return User.findOne({ username: value })
+          .then(userDoc => {
+            if (userDoc) {
+              return Promise.reject('This username is already taken.');
+            }
+          });
+      }),
+    body('password')
+      .trim()
+      .not()
+      .isEmpty()
+      .withMessage('Password cannot be empty.')
+      .isLength({ min: 6 })
+      .withMessage('Password must be at least 6 characters long.')
```

Comment:
I think we should set the minimum password length to 8 and add two more rules:
* must contain at least a number
* must contain at least an uppercase char